### PR TITLE
Add initial game grid

### PR DESCRIPTION
**Pull request by gianbelinche**
Added game logic, this include:

Creating an initial grid, rigth now it has a harcoded example, in the future it will accept a file as input

Updating the grid, given a game grid, it returns the next grid state following the conway's game of life rules

Expected execution:

Captura de pantalla 2024-04-09 a la(s) 11 07 40 a  m
Captura de pantalla 2024-04-09 a la(s) 11 08 17 a  m
Captura de pantalla 2024-04-09 a la(s) 11 07 47 a  m

(Continues indefinitely)

### CODE REVIEW

File: src/game_logic.rs [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    if (i + 1) > (squares - 1) as i32 {
+        0
+    } else {
+        i + 1
+    }
+}
+
+
+mod tests {
+    use super::update_game_grid;
```
Comment (isoprophlex) [inline src/game_logic.rs:106]: Unused Import. It throws a Warning

File: src/game_logic.rs [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    use super::GameGrid;
```
Comment (isoprophlex) [inline src/game_logic.rs:107]: Same as above.

File: src/game_logic.rs [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+    use super::State;
```
Comment (isoprophlex) [inline src/game_logic.rs:108]: It throws a Warning too.

**Review by isoprophlex — CHANGES_REQUESTED** [review]

File: src/game_logic.rs (hunk 1 above)
Comment (gianbelinche) [reply src/game_logic.rs:106]: Done

File: src/game_logic.rs (hunk 2 above)
Comment (gianbelinche) [reply src/game_logic.rs:107]: Done

File: src/game_logic.rs (hunk 3 above)
Comment (gianbelinche) [reply src/game_logic.rs:108]: Done

**Review by isoprophlex — APPROVED** [review]
Everything is great!

**Pull request merged**